### DISCUSSION TITLE
Exposing the Swagger Spec file from the Swagger Documentation Plugin

### DISCUSSION THREAD

**Discussion by alchemycs** [strapi/strapi#8844] [Legacy Topics]
I'm interested in what people thoughts are on this.

I have a need to allow third parties access to the generated swagger JSON spec file, so thought I would take a look at the plugin so it could be exposed, however I found this comment:

> We don't expose the specs using koa-static or something else due to security reasons.
> That's why, we need to read the file localy and send the specs through it when we serve the Swagger UI.
[packages/strapi-plugin-documentation/controllers/Documentation.js#L39](https://github.com/TacticalLimit/strapi/blob/aa26bd95cdeaaf80b65e4fdeda9c2c04b3f86eb0/packages/strapi-plugin-documentation/controllers/Documentation.js#L39)

Although this comment claims it isn't exposed for 'security reasons' it then goes on to expose it anyway in the swagger ui template: [packages/strapi-plugin-documentation/public/index.html#L36](https://github.com/TacticalLimit/strapi/blob/aa26bd95cdeaaf80b65e4fdeda9c2c04b3f86eb0/packages/strapi-plugin-documentation/public/index.html#L36).

Anyone could just view the source of the ui and copy the JSON spec anyway.

If anyone is concerned that exposing the swagger spec is a security risk, then it can be protected through policies the same way the swagger UI is: [packages/strapi-plugin-documentation/config/routes.json#L9](https://github.com/TacticalLimit/strapi/blob/aa26bd95cdeaaf80b65e4fdeda9c2c04b3f86eb0/packages/strapi-plugin-documentation/config/routes.json#L9)

With that in mind I've already created a fork to allow the spec to to be retrieved with the same policy as the swagger ui, and I'm happy to do a PR, but would like to hear people's thoughts before doing so.

What do you think?

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
